File: api/send_link.py
import json
import sys
sys.path.insert(0, '..')

from amprem_client import send_link

def handler(request):
    try:
        if request.method != 'POST':
            return {
                'statusCode': 405,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*',
                    'Access-Control-Allow-Methods': 'POST, OPTIONS',
                    'Access-Control-Allow-Headers': 'Content-Type'
                },
                'body': json.dumps({'status': False, 'message': 'Method not allowed'})
            }
        
        if request.method == 'OPTIONS':
            return {
                'statusCode': 200,
                'headers': {
                    'Access-Control-Allow-Origin': '*',
                    'Access-Control-Allow-Methods': 'POST, OPTIONS',
                    'Access-Control-Allow-Headers': 'Content-Type'
                },
                'body': ''
            }
        
        body = {}
        try:
            if hasattr(request, 'body') and request.body:
                if isinstance(request.body, str):
                    body = json.loads(request.body)
                elif isinstance(request.body, dict):
                    body = request.body
        except Exception as e:
            print(f'Body parse error: {e}')
            body = {}
        
        email = body.get('email', '').strip()
        
        if not email or '@' not in email:
            return {
                'statusCode': 400,
                'headers': {
                    'Content-Type': 'application/json',
                    'Access-Control-Allow-Origin': '*'
                },
                'body': json.dumps({
                    'status': False,
                    'message': 'Email tidak valid atau kosong',
                    'debug': {'email': email or '(empty)', 'body': str(body)}
                })
            }
        
        print(f'Sending link to: {email}')
        result = send_link(email)
        print(f'Result: {result}')
        
        return {
            'statusCode': 200 if result.get('status') else 400,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps(result)
        }
        
    except Exception as e:
        print(f'send-link error: {e}', exc_info=True)
        return {
            'statusCode': 500,
            'headers': {
                'Content-Type': 'application/json',
                'Access-Control-Allow-Origin': '*'
            },
            'body': json.dumps({
                'status': False,
                'message': str(e) or 'Gagal mengirim link'
            })
        }

File: api/send-link.js
const AmPrem = require('amprem');

function parseBody(req) {
  try {
    if (typeof req.body === 'string') {
      return JSON.parse(req.body);
    }
    if (req.body && typeof req.body === 'object') {
      return req.body;
    }
    return {};
  } catch {
    return {};
  }
}

async function withRetry(fn, retries = 2, delay = 1000) {
  for (let i = 0; i <= retries; i++) {
    try {
      return await fn();
    } catch (error) {
      if (i === retries) throw error;
      await new Promise((resolve) => setTimeout(resolve, delay));
    }
  }
}

module.exports = async function handler(req, res) {
  if (req.method !== 'POST') {
    return res.status(405).json({ status: false, message: 'Method not allowed' });
  }

  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  const body = parseBody(req);
  const { email } = body;

  if (!email || !email.includes('@')) {
    return res.status(400).json({
      status: false,
      message: 'Email tidak valid atau kosong'
    });
  }

  try {
    const response = await withRetry(() => AmPrem.sendLink(email));

    if (typeof response === 'string') {
      return res.status(400).json({
        status: false,
        message: response
      });
    }

    return res.status(200).json(response);
  } catch (error) {
    console.error('send-link error:', error);
    return res.status(500).json({
      status: false,
      message: error.message || 'Gagal mengirim link'
    });
  }
};
